sklearn/metrics/scorer.py

```diff
@@ -49,6 +50,16 @@ def __init__(self, score_func, sign, kwargs):
         self._kwargs = kwargs
         self._score_func = score_func
         self._sign = sign
+        # XXX After removing the deprecated scorers (v0.24) remove the
```

Comment:
Why are we putting this deprecation logic into `_BaseScorer` rather and its `__call__`, rather than into `get_scorer`?

Reply:
Are there many other scorers that we need to deprecate?
Else, it seems that we're changing the all the classes just for the sake of the brier scorer.

Reply:
I don't mind having some of this infrastructure here just in case we need it agaijn...

Reply:
that's the way we did it when we introduced all the ``neg`` things right?

Reply:
> that's the way we did it when we introduced all the neg things right?

yes, @amueller see #7261